tests/test_kernel_language.py

```diff
@@ -186,6 +187,31 @@ def kernel_pnum(particle, fieldset, time):
         assert error_thrown
 
 
+def test_varname_as_fieldname():
```

Comment:
You can use `pytest.raises(Exception)`. It is a much easier way to test that the code raise exception.

See also: https://stackoverflow.com/questions/23337471/how-do-i-properly-assert-that-an-exception-gets-raised-in-pytest